File: 2024/Golang/day2/part1.go
package day2

import (
	"Golang/utils"
	"bufio"
	"fmt"
	"os"
	"strings"
)

func Part1() {
	count := 0
	file, err := os.Open("../Input/day2.1.txt")
	utils.Check(err)
	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		numbers := strings.Split(line, " ")
		report, err := utils.StrSliceToIntSlice(numbers)
		if err != nil {
			fmt.Println("Error parsing numbers:",err)
			continue
		}

		if utils.IsSafe(report) {
			count++
		}
	}
	fmt.Println(count)
}

File: 2024/Rust/day2/part1.rs
use crate::utils;

pub fn run_part1() {
    let mut safe_count = 0;
    for line in utils::read_lines("../Input/day2.1.txt") {
        let report: Vec<i64> = line.split(' ').map(|x| x.parse::<i64>().unwrap()).collect();
        if utils::is_safe(report) {
            safe_count += 1;
        }
    }

    println!("{}", safe_count);
}